megamek/src/megamek/client/bot/princess/Princess.java

```diff
@@ -624,7 +624,13 @@ protected void calculateFiringTurn() {
                     }
     
                     // tell the game I want to fire
-                    sendAttackData(shooter.getId(), plan.getEntityActionVector());
+                    Vector<EntityAction> actions = plan.getEntityActionVector();
+                    EntityAction spotAction = getFireControl(shooter).getSpotAction(plan, shooter, fireControlState);
```

Comment:
`getFireControl` is always non-null?

Reply:
Yeah, they get initialized whenever a Princess first loads up. The default is the standard FireControl.